docs/understand/cmake_packages.rst

```diff
@@ -1,139 +1,188 @@
-===========================
+***********
 Using CMake
-===========================
+***********
 
-Most components in ROCm support CMake 3.5 or higher out-of-the-box and do not
-require any special Find modules. A Find module is often used by downstream to
-find the files by guessing locations of files with platform-specific hints.
-Typically, the Find module is required when the upstream is not built with CMake
-or the package configuration files are not available.
+Most components in ROCm support CMake. Projects depnding on header-only or
+library components typically require CMake 3.5 or higher whereas those wanting
+to make use of CMake's HIP language support will require CMake 3.21 or higher.
+
+Finding Dependencies
+====================
+
+.. note::
+   For a complete
+   reference on how to deal with dependencies in CMake, refer to the CMake docs
+   on `find_package
+   <https://cmake.org/cmake/help/latest/command/find_package.html>`_ and the
+   `Using Dependencies Guide
+   <https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html>`_
+   to get an overview of CMake's related facilities.
+
+In short, CMake supports finding dependencies in two ways:
+
+-  In Module mode, it consults a file ``Find<PackageName>.cmake`` which tries to
+   find the component in typical install locations and layouts. CMake ships a
+   few dozen such scripts, but users and projects may ship them as well.
+-  In Config mode, it locates a file named ``<packagename>-config.cmake`` or
+   ``<PackageName>Config.cmake`` which describes the installed component in all
+   regards needed to consume it.
+
+ROCm dominantly relies on Config mode, one notable exception being the Module
+driving the compilation of HIP programs on Nvidia runtimes. As such, when
```

Comment:
I feel as if the difference between these two would be more clear with a simple hello-world (e.g., do a printf from the GPU) for both the AMD and NV backends to see how the CMake would differ.  I'm happy to defer that to a subsequent issue though.

Reply:
Let's add this request to rocm-examples

Reply:
Once when example is made, we can extract the relevant parts here, or just link to the sample. My hope is that eventually we can retire that Module by adding its functionality to the HIP language support.

Just look at any of the HIP-Basic samples of ROCm-examples ([saxpy](https://github.com/amd/rocm-examples/blob/develop/HIP-Basic/saxpy/CMakeLists.txt) for eg.) or any of the HIP library samples that also have device code (hipCUB [device_sum](https://github.com/amd/rocm-examples/blob/develop/Libraries/hipCUB/device_sum/CMakeLists.txt) for eg.). They all properly support both HIP on AMD and HIP on NV and the build definition is a series of jumping through flaming hoops. I'm certain we could do better than that.